### PR TITLE
Swift 4.1

### PR DESCRIPTION
**Pull request by NachoSoto**
~~I'm using [the latest development snapshot](https://swift.org/builds/swift-4.1-branch/xcode/swift-4.1-DEVELOPMENT-SNAPSHOT-2018-01-18-a/swift-4.1-DEVELOPMENT-SNAPSHOT-2018-01-18-a-osx.pkg) from January 18th. This compiles! 🎉~~

Last updated and tested with: `Xcode 9.3 beta 2`.

### TODO:

- [x] Wait for `Xcode 9.3`
- [ ] Update Travis
- ~~Update build setting (_it's still set to `4.0`, [because the toolchain didn't change the build setting](https://twitter.com/NachoSoto/status/956976888420876288)_)~~
- ~~Other improvements?~~

### CODE REVIEW

**Comment by NachoSoto** [comment]
I'm also [not really sure why we still need the manual `==` implementation](https://twitter.com/NachoSoto/status/954463669747765248). I thought we could get that for free?

File: Result/ResultProtocol.swift [hunk 1]
```diff
@@ -121,12 +121,18 @@ extension Result where Value: Equatable, Error: Equatable {
...
+}
 
-	/// Returns `true` if `left` and `right` represent different cases, or if they represent the same case but different values.
-	public static func !=(left: Result<Value, Error>, right: Result<Value, Error>) -> Bool {
-		return !(left == right)
+#if swift(>=4.1)
+	extension Result: Equatable where Value: Equatable, Error: Equatable { }
+#else
+	extension Result where Value: Equatable, Error: Equatable {
+		/// Returns `true` if `left` and `right` represent different cases, or if they represent the
```
Comment (bencochran) [inline Result/ResultProtocol.swift]: Missing the end of this comment

**Comment by ileitch** [comment]
Anything I can do to help expedite this? Result is blocking our upgrade to 4.1 as all the Swift 4 changes in master are not released.

**Comment by mdiep** [comment]
The current release of Result works with Xcode 9.3 / Swift 4.1. How is this blocking you?

**Comment by ileitch** [comment]
I'm getting this error:
```
error: The following binaries use incompatible versions of Swift:
/path/to/DerivedData/Org-ffzmalvhznxtrccncwsuvfbyfypt/Build/Products/Debug-iphonesimulator/Bowtie.app/Bowtie
/path/to/DerivedData/Org-ffzmalvhznxtrccncwsuvfbyfypt/Build/Products/Debug-iphonesimulator/Bowtie.app/PlugIns/BowtieUnitTests.xctest/Frameworks/Result.framework/Result
```

**Comment by NachoSoto** [comment]
FYI I’m out of town for a few weeks so somebody else will need to pick this up from me.

**Comment by ileitch** [comment]
After pointing our Podfile at master the build succeeds.

**Comment by ikesyo** [comment]
I pick this up.

File: Result/ResultProtocol.swift (hunk 1 above)
Comment (ikesyo) [reply Result/ResultProtocol.swift]: Thanks!

**Comment by mdiep** [comment]
Should we wait for an updated Travis image?

**Comment by mdiep** [comment]
Travis has released the `xcode9.3` image, so this just needs to be updated now.

**Comment by ikesyo** [comment]
Just updated to use `xcode9.3` image.

**Comment by NachoSoto** [comment]
👏🏼

**Pull request merged**